src/PolygonClipper/StablePriorityQueue{T,TComparer}.cs

```diff
@@ -39,108 +47,124 @@ public StablePriorityQueue(TComparer comparer)
     /// Adds an item to the priority queue, maintaining the heap property.
     /// </summary>
     /// <param name="item">The item to add.</param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public void Enqueue(T item)
     {
-        this.heap.Add(item);
-        this.Up(this.heap.Count - 1);
+        if (this.count == this.heap.Length)
+        {
+            this.Resize();
+        }
+
+        // Direct array access without bounds checking
+        Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(this.heap), this.count) = item;
+        this.Up(this.count);
+        this.count++;
     }
 
     /// <summary>
     /// Removes and returns the item with the highest priority (lowest value) from the priority queue.
     /// </summary>
     /// <returns>The item with the highest priority.</returns>
     /// <exception cref="InvalidOperationException">Thrown if the priority queue is empty.</exception>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public T Dequeue()
     {
-        if (this.heap.Count == 0)
+        if (this.count == 0)
         {
             throw new InvalidOperationException("Queue is empty.");
         }
 
-        T top = this.heap[0];
-        T bottom = this.heap[^1];
-        this.heap.RemoveAt(this.heap.Count - 1);
+        ref T heapRef = ref MemoryMarshal.GetArrayDataReference(this.heap);
+        T top = heapRef; // Get root element
+        this.count--;
 
-        if (this.heap.Count > 0)
+        if (this.count > 0)
         {
-            this.heap[0] = bottom;
+            // Move last element to root
+            heapRef = Unsafe.Add(ref heapRef, this.count);
+
+            // Clear the last position to avoid holding references
+            Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(this.heap), this.count) = default(T)!;
             this.Down(0);
         }
-
-        return top;
-    }
-
-    /// <summary>
-    /// Returns the item with the highest priority (lowest value) without removing it.
-    /// </summary>
-    /// <returns>The item with the highest priority.</returns>
-    /// <exception cref="InvalidOperationException">Thrown if the priority queue is empty.</exception>
-    public T Peek()
-    {
-        if (this.heap.Count == 0)
+        else
         {
-            throw new InvalidOperationException("Queue is empty.");
+            // Clear the last remaining element
+            heapRef = default(T)!;
         }
 
-        return this.heap[0];
+        return top;
     }
 
     /// <summary>
     /// Restores the heap property by moving the item at the specified index upward.
     /// </summary>
     /// <param name="index">The index of the item to move upward.</param>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private void Up(int index)
     {
-        List<T> data = this.heap;
-        T item = data[index];
+        ref T heapRef = ref MemoryMarshal.GetArrayDataReference(this.heap);
+        T item = Unsafe.Add(ref heapRef, index);
```

Comment:
Here and elsewhere `Unsafe.Add` produces poor codegen for `int`. You should use `uint` or `nuint`